iModelCore/iModelPlatform/DgnCore/DgnDbSchema.cpp

```diff
@@ -546,8 +546,17 @@ ProfileState DgnDb::_CheckProfileVersion() const
 /*---------------------------------------------------------------------------------**//**
 * @bsimethod
 +---------------+---------------+---------------+---------------+---------------+------*/
-BeSQLite::DbResult DgnDb::_OnBeforeProfileUpgrade()
+BeSQLite::DbResult DgnDb::_OnBeforeProfileUpgrade(Db::OpenParams const& params)
     {
+    if (GetBriefcaseId().IsBriefcase())

```

Comment:
what you really want to test here is whether the iTwinId is valid - if not, that means it's a "standalone" iModel.

Reply:
actually, i think you should remove this test entirely. If someone wants to use standalone iModels, they should pass `true` for `schemaLockHeld`

Reply:
I removed it.